source/vir/src/ast_to_sst.rs

```diff
@@ -657,8 +690,44 @@ fn stm_call(
     dest: Option<Dest>,
 ) -> Result<Stm, VirErr> {
     let fun = get_function(ctx, span, &name)?;
-    let mut small_args: Vec<Exp> = Vec::new();
     let mut stms: Vec<Stm> = Vec::new();
+    if ctx.expand_flag && crate::split_expression::need_split_expression(ctx, span) {
+        // we are spliting the `requires` expression on the call site.
+        // If we split the `requires` expression on the function itself,
+        // this splitted encoding will take effect on every call site, which is not desirable.
+        //
+        // Also, note that prevasive::assert is consisted of `requires` and `ensures`.
+        // therefore, we are also splitting pervaisve::assert here
+        let params = &fun.x.params;
+        for e in &**fun.x.require {
+            let exp = crate::split_expression::pure_ast_expression_to_sst(ctx, e, params);
+            let arg_exps = Arc::new(vec_map(&args, |x| x.0.clone()));
```

Comment:
This should use `small_args` rather than `args`.

Reply:
I thought I need the whole `args` to be able to split the arguments too. (If the argument of `assert` can be split, then converting `assert(A && B)` to `assert(A)` and `assert(A => B)`).
but I might be missing obvious facts here; could you explain?

Reply:
Oh, I see.  Never mind, then.  Using `args` sounds like the right thing.

Reply:
thank you!